src/sentry/sentry_metrics/client/kafka.py

```diff
@@ -78,12 +79,17 @@ def counter(
         produced to the broker yet.
         """
 
+        if timestamp is not None:
+            payload_timestamp = timestamp
```

Comment:
Do we need to validate that the timestamp is sane here?

Reply:
Hm, that's a good point. I wonder if we have any downstream checks for this already (in Snuba?). 

Moreover, we might want to align on what is considered okay and what is not

Reply:
I'm planning on enforcing the same boundaries we have in Relay here (-5d, +1min)

Reply:
> Hm, that's a good point. I wonder if we have any downstream checks for this already (in Snuba?).

I think snuba calls this function
https://github.com/getsentry/snuba/blob/7008d0820dbe22142e646c793f7154c96bce0f9b/snuba/processor.py#L189-L195C46
I'm not sure what needs to be done here tho, because I don't think I understand that code fully

Reply:
That's dealing with a `datetime` instead of a `timestamp` though. I think your check Riya makes sense. You might also want to cast it to an int just in case.